crates/bench/benches/callgrind.rs

```diff
@@ -0,0 +1,681 @@
+/// Benchmarks that run under our iai-callgrind fork (https://github.com/clockworklabs/iai-callgrind)
+/// Callgrind (https://valgrind.org/docs/manual/cl-manual.html) disassembles linux binaries
+/// as they run to insert instrumentation code. This allows for benchmarking with minimal variance,
+/// compared to timing benchmarks.
+///
+/// Note: you CAN'T save state between these benchmarks, since each is run
+/// in a fresh process.
+///
+/// FIXME(jgilles): many of the spacetime_module benchmarks are currently disabled due to a bad interaction with sled (issue #564).
+///
+/// There is some odd boilerplate in this file that is used to make downstream processing of these benchmarks easier.
```

Comment:
I happen to find this strategy pretty clear. In particular, once set up, it is easy for me to imagine someone other than James adding a record for a new benchmark, which would be my most preferred thing to optimize for. I do think others should probably weigh in on this point.